packages/react-native-codegen/src/parsers/parsers-primitives.js

```diff
@@ -214,6 +224,70 @@ function emitFloat(
   });
 }
 
+function translateArrayTypeAnnotation(
+  hasteModuleName: string,
+  types: TypeDeclarationMap,
+  aliasMap: {...NativeModuleAliasMap},
+  cxxOnly: boolean,
+  arrayType: 'Array' | 'ReadonlyArray',
+  elementType: $FlowFixMe,
+  nullable: boolean,
+  language: ParserType,
+): Nullable<NativeModuleTypeAnnotation> {
+  try {
+    /**
+     * TODO(T72031674): Migrate all our NativeModule specs to not use
+     * invalid Array ElementTypes. Then, make the elementType a required
+     * parameter.
+     */
+    const [_elementType, isElementTypeNullable] = unwrapNullable(
+      language === 'TypeScript'
+        ? typeScriptTranslateTypeAnnotation(
+            hasteModuleName,
+            elementType,
+            types,
+            aliasMap,
+            /**
+             * TODO(T72031674): Ensure that all ParsingErrors that are thrown
+             * while parsing the array element don't get captured and collected.
+             * Why? If we detect any parsing error while parsing the element,
+             * we should default it to null down the line, here. This is
+             * the correct behaviour until we migrate all our NativeModule specs
+             * to be parseable.
+             */
+            nullGuard,
+            cxxOnly,
+          )
+        : flowTranslateTypeAnnotation(
+            hasteModuleName,
+            elementType,
+            types,
+            aliasMap,
+            nullGuard,
+            cxxOnly,
+          ),
```

Comment:
Can we add as parameter of this function a lambda:
```ts
translateTypeAnnotation: (
  hasteModuleName: string,
  elementType: $FlowFixMe,
  types: TypeDeclarationMap,
  aliasMap: {...NativeModuleAliasMap},
  nullGuard: <add type>
  cxxOnly: boolean
) => [$FlowFixMe, boolean]
```

and:
1. avoid the `require` of the two functions (which is a code smell, imho)
2. avoid to run a ternary operation
?

this should simplify the code also because here you would just invoke the lambda you received, saving ~10 LOC.

Reply:
the exports tricked me there.. haha!
will push the refac. by tmrw
Thanks :)

Reply:
\#offtopic
*Ooops forgot to say; Nice DP!*

Reply:
> translateTypeAnnotation: (
  ...
) => `[$FlowFixMe, boolean]`

This return type is wrong.

---

Pushed the changes. Thanks 👍